device/tt_device.h

```diff
@@ -785,7 +786,7 @@ class tt_SiliconDevice: public tt_device
     std::set<chip_id_t> target_remote_chips = {};
     tt_SocDescriptor& get_soc_descriptor(chip_id_t chip_id);
     tt::ARCH arch_name;
-    std::map<chip_id_t, struct PCIdevice*> m_pci_device_map;    // Map of enabled pci devices
+    std::map<chip_id_t, std::unique_ptr<PCIDevice>> m_pci_device_map;    // Map of enabled pci devices
```

Comment:
I wonder if the performance of `unordered_map` is any better here?

Reply:
I think it's being used basically as a look up table so performance for `unordered_map` is probably better. 

Changed.

Reply:
I think that map is better for small containers (<1000), due to a large constant involved due to hashing. unordered_map generally has a bad standard implementation. But let's not think about it too much, either should be fast enough for now. And this PR is not about perf anyway